tensorflow/compiler/xla/service/gpu/cudnn_fused_conv_rewriter_test.cc

```diff
@@ -142,6 +142,45 @@ class CudnnFusedConvRewriterTest : public GpuCodegenTest {
                   Not(HasSubstr(kCudnnConvBiasActivationForwardCallTarget)));
     }
   }
+
+  void TestF8(absl::string_view pre_hlo_string,
+              absl::string_view custom_call_string,
+              absl::string_view serialized_graph_string) {
+    if (GetCudaComputeCapability().IsAtLeast(
+            se::CudaComputeCapability::HOPPER)) {
```

Comment:
Isn't that the solution we arrived at? As I understand it, you don't want to verify the final layout even on Hopper systems which in my opinion renders the test somewhat incomplete. I don't think this is directly comparable to the GEMM case where layout conversions play less of a role.

Reply:
I'm still unsure why layout is important here, compared to the gemm case. Running GpuConvRewriter and CudnnFusedConvRewriter on `pre_hlo_string` is still causing the FP8 rewrite to happen even if layout assignment doesn't run, right?

Even layout assignment is important, maybe see what transformations it is doing to `pre_hlo_string`, and just putting the resulting layouts directly in the HLO strings in the test, instead of relying on layout assignment to run.

Reply:
My perspective is that the XLA unit tests are usually based on running the compiler pipeline. When that's not possible, we can do at partial testing by running only the relevant pass in some artificial setting that we can't easily extend to the full pipeline. It's less clear to me though why we'd want to deviate from the normal approach and restrict the testing in cases where we don't have to as well.

Reply:
The reason to deviate from the normal approach is that we want to test as much as possible on non-Hopper, and right now, the PR only tests the graph string and not things like the custom_call_target, the dim_labels, etc.

But I'll accept only testing the graph string for now, we can reconsider if the tests get broken later due to a lack of Hopper CI.

Reply:
We can discard the first part of the custom_call FileCheck string on non-Hopper systems and still compare the configuration of the Custom Call.

Reply:
What is the first part of the custom_call FileCheck string? Is this the `f8e4m3fn[1,6,6,16]{3,2,1,0}, u8[{{.*}}]{0})` part and is that part different if you don't run the rest of the passes?

Reply:
Yes, the order of the dimensions changes.